core/src/main/java/com/github/jsonldjava/utils/JsonUtils.java

```diff
@@ -371,6 +363,56 @@ public static Object fromURL(java.net.URL url, CloseableHttpClient httpClient)
         }
     }
 
+    private static InputStream getJsonLdViaHttpUri(final URL url, final CloseableHttpClient httpClient,
+            CloseableHttpResponse response) throws IOException {
+        final HttpUriRequest request = new HttpGet(url.toExternalForm());
+        // We prefer application/ld+json, but fallback to application/json
+        // or whatever is available
+        request.addHeader("Accept", ACCEPT_HEADER);
+        response = httpClient.execute(request);
+
+        final int status = response.getStatusLine().getStatusCode();
+        if (status != 200 && status != 203) {
+            throw new IOException("Can't retrieve " + url + ", status code: " + status);
+        }
+        // follow alternate document location
+        // https://www.w3.org/TR/json-ld11/#alternate-document-location
+        URL alternateLink = alternateLink(url, response);
+        if (alternateLink != null) {
+            return getJsonLdViaHttpUri(alternateLink, httpClient, response);
+        }
+        return response.getEntity().getContent();
+    }
+
+    private static URL alternateLink(URL url, CloseableHttpResponse response)
+            throws MalformedURLException, IOException {
+        if (response.getEntity().getContentLength() > 0
```

Comment:
Why does the content length need to be greater than 0 if we are only looking through the headers in this case? Is there anything stopping someone returning HTTP 204 (except we don't support it currently above with our hardcoded `200 or 203` check)?

Reply:
Good catch, better to test if a contentType is there - contentLenght may be 0. Treatet with faddb48a65b53e7541359ab0a1a6840e125c4239.